includes/Abilities/Utilities/Utilities.php

```diff
@@ -0,0 +1,250 @@
+<?php
+/**
+ * Utility WordPress Abilities implementation.
+ *
+ * @package WordPress\AI
+ */
+
+declare( strict_types=1 );
+
+namespace WordPress\AI\Abilities\Utilities;
+
+use WP_Error;
+
+/**
+ * Utility WordPress Abilities.
+ *
+ * @since 0.1.0
+ */
+class Utilities {
+
+	/**
+	 * The fields that we support.
+	 *
+	 * @since 0.1.0
+	 * @var array<string>
+	 */
+	private static array $post_details_fields = array( 'content', 'title', 'slug', 'author', 'type', 'excerpt' );
+
+	/**
+	 * Register any needed hooks.
+	 *
+	 * @since 0.1.0
+	 */
+	public function register(): void {
+		add_action( 'wp_abilities_api_init', array( $this, 'register_abilities' ) );
+	}
+
+	/**
+	 * Registers any needed abilities.
+	 *
+	 * @since 0.1.0
+	 */
+	public function register_abilities(): void {
+		$this->register_get_post_details_ability();
+		$this->register_get_terms_ability();
+	}
+
+	/**
+	 * Registers the get-post-details ability.
+	 *
+	 * @since 0.1.0
+	 */
+	private function register_get_post_details_ability(): void {
+		wp_register_ability(
+			'ai/get-post-details',
+			array(
+				'label'               => esc_html__( 'Get post details', 'ai' ),
+				'description'         => esc_html__( 'Get the details of a post based on the post ID. Optionally limit the details to specific fields.', 'ai' ),
+				'category'            => AI_DEFAULT_ABILITY_CATEGORY,
+				'input_schema'        => array(
+					'type'       => 'object',
+					'properties' => array(
+						'post_id' => array(
+							'type'        => 'integer',
+							'description' => esc_html__( 'The ID of the post to get the details of.', 'ai' ),
+						),
+						'fields'  => array(
+							'type'        => 'array',
+							'description' => esc_html__( 'The fields to get the details of. Will default to all fields if not provided.', 'ai' ),
+							'items'       => array(
+								'type' => 'string',
+								'enum' => self::$post_details_fields,
+							),
+						),
+					),
+					'required'   => array( 'post_id' ),
+				),
+				'output_schema'       => array(
+					'type'       => 'object',
+					'properties' => array(
+						'details' => array(
+							'type'        => 'array',
+							'description' => esc_html__( 'An array of post details.', 'ai' ),
+						),
```

Comment:
It doesn't look like it's returning an `array( 'details' => array( ... ) )`, but instead an array tentatively containing `content`, `title`, and such. I think we should list out each parameter, too, as that especially tells systems like AI models what exactly to expect. For example, an AI might assume it's receiving `post_content` instead of `content`.